src/ledger/LedgerTxnEntry.cpp

```diff
@@ -83,7 +84,7 @@ LedgerTxnEntry::Impl::~Impl()
 }
 
 LedgerTxnEntry::LedgerTxnEntry(LedgerTxnEntry&& other)
-    : mImpl(std::move(other.mImpl))
+    : mImpl(std::move(other.mImpl)), mIsInitialized(true)
```

Comment:
I don't think this is wrong today, but it is a foot-gun. I think it should say `mIsInitialized(other.mIsInitialized)`. `LedgerTxnEntry()` is not identical to `LedgerTxnEntry(LedgerTxnEntry())` even though I would expect them to be the same. Why should initializing with something that is not initialized result in something that is initialized?

Reply:
Good catch. I updated the initialization of `mIsInitialized` in `LedgerTxnEntry` and `ConstLedgerTxnEntry` and added test cases in a fixup commit.